src/elements/content-explorer/ContentExplorer.js

```diff
@@ -388,16 +388,18 @@ class ContentExplorer extends Component<Props, State> {
     fetchFolderSuccessCallback(collection: Collection, triggerNavigationEvent: boolean): void {
         const { onNavigate, rootFolderId }: Props = this.props;
         const { id, name, boxItem }: Collection = collection;
+        const { selected }: State = this.state;
 
         // New folder state
         const newState = {
+            currentCollection: { ...collection },
             selected: undefined,
-            currentCollection: collection,
             rootName: id === rootFolderId ? name : '',
         };
 
-        // Unselect any rows that were selected
-        this.unselect();
+        const { newCollection, newSelected } = this.updateCollection(collection, selected);
+        newState.currentCollection = newCollection;
+        newState.selected = newSelected;
```

Comment:
I think you could move L400 above and then declare the `newState` one time instead of doing these reassignments